File: CS/DXApplication5/Form1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using DevExpress.XtraScheduler;
using DevExpress.Services;



namespace DXApplication5
{
    public partial class Form1 : XtraForm
    {


        private void InitSchedulerControl()
        {
            schedulerControl.Start = System.DateTime.Now;
            schedulerControl.ActiveViewType = SchedulerViewType.Day;
            schedulerControl.GroupType = SchedulerGroupType.Resource;
            schedulerControl.DayView.DayCount = 1;
            schedulerControl.DayView.ResourcesPerPage = 2;
            schedulerControl.OptionsView.EnableAnimation = false;
       
        }
        public Form1()
        {
            InitializeComponent();
            InitSchedulerControl();

            schedulerStorage.Appointments.ResourceSharing = true;

            DataHelper.FillResources(schedulerStorage, 5);
            InitAppointments();
        }


        void InitAppointments()
        {
            int count = schedulerStorage.Resources.Count;
            DataHelper.GenerateEvents(schedulerStorage.Appointments.Items, count, schedulerStorage.Resources.Items);
            DataHelper.GenerateEventsWithSharedResource(schedulerStorage.Appointments.Items, 1, schedulerStorage.Resources.Items);
            DataHelper.AddRecurrentAppointment(schedulerStorage.Appointments.Items);
        }

        private void ExchangeMouseService()
        {
            IMouseHandlerService oldMouseHandler = schedulerControl.GetService<IMouseHandlerService>();
            if (oldMouseHandler != null)
            {
                MyMouseHandlerService newMouseHandler = new MyMouseHandlerService(schedulerControl, oldMouseHandler);
                schedulerControl.RemoveService(typeof(IMouseHandlerService));
                schedulerControl.AddService(typeof(IMouseHandlerService), newMouseHandler);
            }
        }
    
        private void Form1_Load(object sender, EventArgs e)
        {
            ExchangeMouseService();
        }

    }

}

File: VB/SchedulerControlCustomZooming/DXApplication5/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors
Imports DevExpress.XtraScheduler
Imports DevExpress.Services



Namespace DXApplication5
	Partial Public Class Form1
		Inherits XtraForm


		Private Sub InitSchedulerControl()
			schedulerControl.Start = DateTime.Now
			schedulerControl.ActiveViewType = SchedulerViewType.Day
			schedulerControl.GroupType = SchedulerGroupType.Resource
			schedulerControl.DayView.DayCount = 1
			schedulerControl.DayView.ResourcesPerPage = 2
			schedulerControl.OptionsView.EnableAnimation = False

		End Sub
		Public Sub New()
			InitializeComponent()
			InitSchedulerControl()

			schedulerStorage.Appointments.ResourceSharing = True

			DataHelper.FillResources(schedulerStorage, 5)
			InitAppointments()
		End Sub


		Private Sub InitAppointments()
			Dim count As Integer = schedulerStorage.Resources.Count
			DataHelper.GenerateEvents(schedulerStorage.Appointments.Items, count, schedulerStorage.Resources.Items, schedulerStorage)
			DataHelper.GenerateEventsWithSharedResource(schedulerStorage.Appointments.Items, 1, schedulerStorage.Resources.Items, schedulerStorage)
			DataHelper.AddRecurrentAppointment(schedulerStorage.Appointments.Items, schedulerStorage)
		End Sub

		Private Sub ExchangeMouseService()
			Dim oldMouseHandler As IMouseHandlerService = schedulerControl.GetService(Of IMouseHandlerService)()
			If oldMouseHandler IsNot Nothing Then
				Dim newMouseHandler As New MyMouseHandlerService(schedulerControl, oldMouseHandler)
				schedulerControl.RemoveService(GetType(IMouseHandlerService))
				schedulerControl.AddService(GetType(IMouseHandlerService), newMouseHandler)
			End If
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			ExchangeMouseService()
		End Sub

	End Class

End Namespace